src/orange/OrangeTypes.hh

```diff
@@ -48,6 +54,16 @@ using UniverseId = OpaqueId<struct Universe>;
 //! Opaque index for "simple unit" data
 using SimpleUnitId = OpaqueId<struct SimpleUnitRecord>;
 
+//! Data to specify an embeded universe withing a volume
+struct Daughter
+{
+    UniverseId universe_id;
+    TranslationId translation_id;
+};
+
+//! Opaque index for "simple unit" data
+using DaughterId = OpaqueId<struct Daughter>;
```

Comment:
As with `SimpleUnitRecord`, you can rely on the `struct` inside the template parameter to forward-declare the class, and move the `Daughter` definition down with the other structs. Could you then also keep the IDs here alphabetically sorted?